npm/ipsm/ipsm.go

```
@@ -0,0 +1,439 @@
+func (ipsMgr *IpsetManager) IncrementReferCount(setName string) error {
+	if _, exists := ipsMgr.setMap[setName]; !exists {
+		log.Printf("set %s doesn't exist, can't increment", setName)
+		return fmt.Errorf("set %s doesn't exist, can't increment", setName)
+	}
+
+	ipsMgr.setMap[setName].referCount++
+
+	return nil
+}
+
+// DecrementReferCount decreases referCount of a specific ipset by one.
+func (ipsMgr *IpsetManager) DecrementReferCount(setName string) error {
+	if _, exists := ipsMgr.setMap[setName]; !exists {
+		log.Printf("set %s doesn't exist, can't increment", setName)
+		return fmt.Errorf("set %s doesn't exist, can't increment", setName)
+	}
+
+	ipsMgr.setMap[setName].referCount--
+
+	return nil
+}
+
+// NotReferredByNwPolicy checks if a specific ipset is referred by any network policy.
+func (ipsMgr *IpsetManager) NotReferredByNwPolicy(setName string) bool {
+	return ipsMgr.setMap[setName].referCount == 0
+}
+
+// CreateList creates an ipset list. npm maintains one setlist per namespace label.
+func (ipsMgr *IpsetManager) CreateList(listName string) error {
+	if _, exists := ipsMgr.listMap[listName]; exists {
+		return nil
+	}
+
+	entry := &ipsEntry{
+		name:          listName,
+		operationFlag: util.IpsetCreationFlag,
+		set:           util.GetHashedName(listName),
+		spec:          util.IpsetSetListFlag,
+	}
+	log.Printf("Creating List: %+v\n", entry)
+	if _, err := ipsMgr.Run(entry); err != nil {
+		log.Printf("Error creating ipset list %s.\n", listName)
+		return err
+	}
+
+	ipsMgr.listMap[listName] = NewIpset(listName)
+
+	return nil
+}
+
+// DeleteList removes an ipset list.
+func (ipsMgr *IpsetManager) DeleteList(listName string) error {
+	entry := &ipsEntry{
+		operationFlag: util.IpsetDestroyFlag,
+		set:           util.GetHashedName(listName),
+	}
+
+	errCode, err := ipsMgr.Run(entry)
+	if errCode == 1 && err != nil {
+		log.Printf("Cannot delete list %s as it's being referred or doesn't exist.\n", listName)
+		return nil
```

Why are we retuning nil here? Shouldn't we return the err ?